tests/api_fastapi/execution_api/routes/test_task_instances.py

```diff
@@ -655,6 +655,40 @@ def test_ti_update_state_to_failed_without_retry_table_check(self, client, sessi
         assert ti.next_kwargs is None
         assert ti.duration == 3600.00
 
+    @pytest.mark.parametrize(
+        ("state", "expected_status_code"),
+        [
+            (State.RUNNING, 204),
+            (State.SUCCESS, 409),
+            (State.QUEUED, 409),
+            (State.FAILED, 409),
```

Comment:
Is it possible to test the 400 error case as well?

Reply:
Yeah, can do.

Reply:
Oh I was meant to do it, hence the name of the test is `test_ti_runtime_checks_success`